src/components/RightSidebar/Participants/ParticipantsList/Participant/Participant.vue

```diff
@@ -263,6 +300,13 @@ export default {
 	},
 
 	methods: {
+		handleResize() {
+			const e = this.$refs.userName
+			this.showUserNameTooltip = (e && e.offsetWidth < e.scrollWidth)
+			const e2 = this.$refs.userStatus
+			this.showUserStatusTooltip = (e2 && e2.offsetWidth < e2.scrollWidth)
+		},
```

Comment:
Is this really needed? I feel like a simple would suffice.
It's fine if we just show it for all sizes. It would avoid the cost of a ResizeObserver as well as increase accessibility :)

Reply:
it looks shitty with short names...

<img width="213" alt="image" src="https://user-images.githubusercontent.com/277525/98360214-b6f21b00-2029-11eb-82db-5a43c174ee80.png">

now, there is another possible, reusable but more complex solution: extend the [Tooltip directive](https://nextcloud-vue-components.netlify.app/#/Directives) to be able to check an element's ellipsization only **on hover** and then decide whether to display or not. Or at least provide some events so that the caller could do the check on their own.

I'm fine removing all the conditionals here if we are fine with the above look...

Reply:
Well titles appear on long-hover only, I think it's fine.
I'm afraid a resizeobserver for each participant just doesn't scale.

Reply:
ok, I've removed the conditions

it's less bad than I think as the hover will only appear if you hove on the text itself, not the whole row, so it's ok

Reply:
Couldn't we just observe the resize one level above? we only need to do it once for all participant components. I think that the bob tooltip looks really bad.

Reply:
Even if we observe in the ParticipantList, we still need to trigger an event to all the participant entries individually so they can perform the tooltip check.

The best solution is to have an on-demand check, basically whenever the user hovers, the check is done, and then the decision is made whether the tooltip appears or not. Ideal would be to have this as an option in the v-tooltip directive.

Reply:
I've raised https://github.com/nextcloud/nextcloud-vue/issues/1567 to address this there.

Reply:
if this is the lib we use there is a chance: https://github.com/Akryum/v-tooltip

I see there are event handlers, I'll see if I can prevent displaying...

Reply:
I checked the library and it seems we cannot use it that way.
See my attempts here: https://github.com/nextcloud/nextcloud-vue/issues/1567#issuecomment-725535802

Now I found another trick that works without touching the lib: within a "mouseover" handler, do the ellipsis check and quickly update the tooltip text afterwards. That seems to work for me. See https://github.com/nextcloud/spreed/pull/4548/commits/40d87afa9046632f1a0d3a9ea26d6776614787ac